india_compliance/income_tax_india/overrides/asset_depreciation_schedule.py

```diff
@@ -1,29 +1,132 @@
 import frappe
 from frappe import _
-from frappe.utils import date_diff
+from frappe.utils import (
+    add_days,
+    add_months,
+    cint,
+    date_diff,
+    flt,
+    get_last_day,
+    getdate,
+    is_last_day_of_the_month,
+)
+from erpnext.accounts.utils import get_fiscal_year
+from erpnext.assets.doctype.asset_depreciation_schedule.asset_depreciation_schedule import (
+    get_asset_depr_schedule_doc,
+    get_default_wdv_or_dd_depr_amount,
+)
 
 
-def get_updated_rate_of_depreciation_for_wdv_and_dd(
-    asset, depreciable_value, fb_row, show_msg=True
+def get_wdv_or_dd_depr_amount(
+    asset,
+    fb_row,
+    depreciable_value,
+    schedule_idx,
+    prev_depreciation_amount,
+    has_wdv_or_dd_non_yearly_pro_rata,
+    asset_depr_schedule,
 ):
+    # As per IT act, if the asset is purchased in the 2nd half of fiscal year, then rate is divided by 2 for the first year
+
+    if not fb_row.finance_book or not frappe.db.get_value(
+        "Finance Book", fb_row.finance_book, "for_income_tax"
+    ):
+        return get_default_wdv_or_dd_depr_amount(
+            asset,
+            fb_row,
+            depreciable_value,
+            schedule_idx,
+            prev_depreciation_amount,
+            has_wdv_or_dd_non_yearly_pro_rata,
+        )
+
+    if not fb_row.daily_prorata_based:
```

Comment:
Customers example for pro rata on a daily basis is a personal preference and not a requirement as per income tax act.